File: Core/Domain/Models/Notification.cs
using Shared.Enums;

namespace Domain.Models
{
    public class Notification : BaseEntity<int>
    {
        public string ApplicationUserId { get; set; } = default!;
        public ApplicationUser ApplicationUser { get; set; } = default!;
        public NotificationType Type { get; set; }
        public string Message { get; set; } = default!;

        /// <summary>
        /// Optional JSON payload to store extra contextual data 
        /// (e.g. related VideoId, PlaylistId, FromUserId, etc.).
        /// </summary>
        public string? DataJson { get; set; }

        public bool IsRead { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: mobile/rescufy/lib/domain/entities/notification.dart
import 'package:equatable/equatable.dart';

class Notification extends Equatable {
  const Notification({
    required this.id,
    required this.title,
    required this.message,
    required this.isRead,
    required this.createdAt,
    this.type,
  });

  final int id;
  final String title;
  final String message;
  final bool isRead;
  final DateTime createdAt;
  final String? type;

  @override
  List<Object?> get props => [id, title, message, isRead, createdAt, type];
}
